src/component.js

```diff
@@ -87,6 +87,37 @@ Component.prototype.forceUpdate = function(callback) {
  */
 Component.prototype.render = Fragment;
 
+/**
+ * @param {import('./internal').VNode} vnode
+ * @param {number | null} [childIndex]
+ */
+export function getDomSibling(vnode, childIndex) {
```

Comment:
I forget - does JS support optimized tail recursion? I think this function could be re-written to be tail-recursive if it isn't already (I forget what exactly qualifies as tail recursive).

Reply:
V8 had it for a while but not anymore, tracking bug: https://bugs.chromium.org/p/v8/issues/detail?id=4698